dd-sdk-android-ndk/src/main/cpp/utils/format-utils.cpp

```diff
@@ -0,0 +1,23 @@
+#include <string>
+#include <stdexcept>
+
+namespace strformat {
+
+    template<typename... Args>
+    std::string format(std::string formatter, Args... args) {
+        using namespace std;
+        const auto formatter_value = formatter.c_str();
+        // first we compute the size of the buffer required to format this string
+        // we will add 1 at the end for the end of string /0 character
+        const size_t size = snprintf(nullptr, 0, formatter_value, args...) + 1;
+        if (size <= 0) {
```

Comment:
`size_t` is unsigned, meaning it cannot be negative, and you are adding 1 to the result, I can guarantee you that size will never be less than 1 (and thus this condition cannot possibly trigger)